src/main/java/com/eu/habbo/habbohotel/rooms/Room.java

```diff
@@ -4873,4 +4877,162 @@ public Collection<RoomUnit> getRoomUnitsAt(RoomTile tile) {
         THashSet<RoomUnit> roomUnits = getRoomUnits();
         return roomUnits.stream().filter(unit -> unit.getCurrentLocation() == tile).collect(Collectors.toSet());
     }
+
+    public List<HabboItem> getAreaHiders(boolean onlyActive) {
+        TIntObjectIterator<HabboItem> iterator = this.roomItems.iterator();
+        List<HabboItem> foundItems = new ArrayList<>();
+        Gson gson = new Gson();
+
+        for (int i = this.roomItems.size(); i > 0; i--) {
+            try {
+                iterator.advance();
+                HabboItem object = iterator.value();
+
+                if (object instanceof InteractionAreaHider && object.getExtradata() != null && object.getExtradata().trim().startsWith("{")) {
+                    try {
+                        InteractionAreaHider.JsonData data = gson.fromJson(object.getExtradata(), InteractionAreaHider.JsonData.class);
+
+                        if (!onlyActive || (onlyActive && data.on == 1)) {
+                            foundItems.add(object);
+                        }
+                    } catch (JsonSyntaxException e) {
+                        LOGGER.error("An error occurred: ", e);
+                    }
+                }
+            } catch (Exception e) {
+                LOGGER.error("An error occurred: ", e);
+            }
+        }
+        return foundItems;
+    }
+
+    public void toggleAreaHiderItemsVisibility() {
```

Comment:
This is not toggling but more updating the state of the room items based on the placed area hiders. Maybe `update` instead of `toggle` suits better, or something else.

Would also appreciate a javadoc comment about what this method does exactly.